builtin/aws/utils/ecs.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+func RegisterTaskDefinition(def *ecs.RegisterTaskDefinitionInput, ecsSvc *ecs.ECS) (*ecs.TaskDefinition, error) {
+	// AWS is eventually consistent so even though we probably created the
+	// resources that are referenced by the task definition, it can error out if
+	// we try to reference those resources too quickly. So we're forced to guard
+	// actions which reference other AWS services with loops like this.
+	var taskOut *ecs.RegisterTaskDefinitionOutput
+	var err error
+	for i := 0; i < 30; i++ {
+		taskOut, err = ecsSvc.RegisterTaskDefinition(def)
+		if err != nil {
+			return nil, err
+		}
+
+		if taskOut != nil && taskOut.TaskDefinition != nil {
+			break
+		}
+
+		// otherwise sleep and try again
```

Comment:
Can we log something here?

Reply:
We could, but we would need to inject in a `log` or `stepgroup` into the function. I wonder if instead a better idea would be to lower the imposed timeout in the `for` loop, to surface any errors quicker. To my knowledge the only retry-able error would be something eventually consistent with a new IAM role. Thoughts?

Reply:
It could also be waiting for a security group or a log group. Even if it's 15 seconds, that combined with slow API responses every iteration could leave someone scratching their head as to what waypoint is up to while it's trapped in this loop. Not a hill I want to die on though.

Reply:
I added a debug log in eaf767d0